### PR TITLE
[#45] Describe the three different Serializers in Advanced Tuning section

### PR DESCRIPTION
**Pull request by smcvb**
This PR adds a more concise description on what `Serializer` implementations there are and how these are configured correctly. Additionally, the section is referred to from several location to add clarity.

### CODE REVIEW

File: part-iii-infrastructure-components/spring-boot-autoconfiguration.md [hunk 1]
```diff
@@ -37,26 +37,46 @@ You can define a separate Serializer to be used to serialize events, by assignin
...
+    @Qualifier("eventSerializer")
+    @Bean
+    public Serializer eventSerializer() {
+        return new JacksonSerializer();
+    }
 }
 ```
 
-When overriding both the default serializer and defining an event serializer, we must instruct Spring that the default serializer is, well, the default:
+Equal to events, you can also customize the Message `Serializer` used by your application. The Message `Serializer` comes into play when your Command and Query message are send from one node to another in a distributed environment. To set a custom `Serializer` for you message you can simply define a `messageSerializer` bean like so:
```
Comment (m1l4n54v1c) [inline part-iii-infrastructure-components/spring-boot-autoconfiguration.md]: ... Query messages are *sent* from one ...

File: part-iv-advanced-tuning/advanced-customizations.md [hunk 2]
```diff
@@ -8,6 +8,39 @@ You can configure additional `ParameterResolver`s by extending the `ParameterRes
...
+3. The Default `Serializer` is in charge of de-/serializing the remainder, being the Tokens, Snapshots and Sagas. These objects are generally not shared between different applications, and most of these classes aren't expected to have some of the getters and setters that are, for example, typically required by Jackson based serializers. A flexible, general purpose serializer like [XStream](http://x-stream.github.io/) is quite suited for this purpose.
+
+By default all three `Serializer` flavors are set to use the `XStreamSerializer`, which internally uses [XStream](http://x-stream.github.io/) to serialize objects to an XML format. XML is a verbose format to serialize to, but XStream has the major benefit of being able to serialize virtually anything. This verbosity is typically fine when storing tokens, sagas or snapshots, but for messages (and specifically events) XML might prove to cost too much due to its serialized size. Thus for optimization reasons you can configure different serializers for your messages. 
+
```
Comment (m1l4n54v1c) [inline part-iv-advanced-tuning/advanced-customizations.md]: Suggestion: I would mention interoperability reasons here also. If some other systems (not necessarily written in Java) wants to read the events, some other format is preferred here.

File: part-iii-infrastructure-components/spring-boot-autoconfiguration.md (hunk 1 above)
Comment (smcvb) [reply part-iii-infrastructure-components/spring-boot-autoconfiguration.md]: Sharp, fixed typo.

File: part-iv-advanced-tuning/advanced-customizations.md (hunk 2 above)
Comment (smcvb) [reply part-iv-advanced-tuning/advanced-customizations.md]: Fair suggestion which is good to take into account. Will make the necessary adjustment.

**Pull request merged**